### PR TITLE
Implementation of DID Peer and it's various types 

### PR DESCRIPTION
**Pull request by forkimenjeckayang**
This PR implements the DID Peer method in the DidPeerMethod class. The following DID types are supported:

did:peer:0 - Method type method0
did:peer:1 - Method type method1
did:peer:2 - Method type method2
did:peer:3 - Method type method3
did:peer:4 - Method type method4

### CODE REVIEW

**Comment by forkimenjeckayang** [comment]
This closses: https://github.com/orgs/adorsys/projects/6?pane=issue&itemId=84522794&issue=adorsys%7Ceudiw-app%7C272

**Comment by hugoib** [comment]
The implementation looks very good! I like it. I would just have 2 small remarks:

Remove some console.log on the tests... Try to minimise their usage (or maybe implement a debug testing method?) So that they don't interfere when we want to just run the tests or in the pipeline

We need a did peer method 2 gen where we input the routing keys, look at this example:
Screenshot 2024-11-06 at 13 24 05
source:
https://github.com/roots-id/didcomm-mediator/blob/main/sample-notebooks/Alice.ipynb

Ticket created for the did:resolver: https://github.com/ADORSYS-GIS/wallet-vc-libs/issues/41

File: packages/multiple-did-identities/src/__tests__/did-methodfactory.spec.ts [hunk 1]
```diff
@@ -19,15 +26,130 @@ describe('DidMethodFactory', () => {
...
+    it('should throw an error for an unsupported DID method', () => {
+      expect(() =>
+        DidMethodFactory.create('unsupported' as DIDMethodName),
+      ).toThrow('Unsupported DID method: unsupported');
+    });
+
+    const peerMethods: PeerGenerationMethod[] = [
+      'method0',
+      'method1',
+      'method2',
```
Comment (hugoib) [inline packages/multiple-did-identities/src/__tests__/did-methodfactory.spec.ts:81]: I suggest we use enums for peer method types. Will avoid string typing issues. 

export enum PeerMethodTypes {
   Method0 = 'method0',
 }

File: packages/multiple-did-identities/src/did-methods/DidPeerMethod.ts [hunk 2]
```diff
@@ -8,7 +32,356 @@ import { DIDMethodName } from './DidMethodFactory';
...
+    const keysArray = [purposepublicKeyMultibaseV, purposepublicKeyMultibaseE];
+    const concatPurposeKeys = concatenateKeyStrings(...keysArray);
+
+    // Define a service endpoint
+    const service: Service[] = [
+      {
+        id: '#didcommmessaging',
+        type: 'DIDCommMessaging',
+        serviceEndpoint: {
+          uri: 'http://example.com/didcomm', // mediator uri endpoint
```
Comment (hugoib) [inline packages/multiple-did-identities/src/did-methods/DidPeerMethod.ts:237]: The method will receive a uri that will go into the uri of serviceEndpoint. Please review.

**Review by hugoib — COMMENTED** [review]
Almost but not quite. Alice does not have a service endpoint itself, but it uses the mediator one.

**Review by hugoib — APPROVED** [review]
Great! Thank you!

**Pull request merged**